baybe/utils/validation.py

```diff
@@ -66,3 +75,51 @@ def validator(self: Any, attribute: Attribute, value: Any) -> None:
 
 non_inf_float = _make_restricted_float_validator(allow_nan=True, allow_inf=False)
 """Validator for non-infinite floats."""
+
+
+def get_transform_objects(
+    objects: Sequence[_T],
+    df: pd.DataFrame,
+    allow_missing: bool,
+    allow_extra: bool,
+) -> list[_T]:
+    """Extract the objects relevant for transforming a given dataframe.
+
+    The passed object are assumed to have corresponding columns in the given dataframe,
+    identified through their name attribute. The function returns the subset of objects
+    that have a corresponding column in the dataframe and thus provide the necessary
+    information for transforming the dataframe.
+
+    Args:
+        objects: A collection of objects to be considered for transformation (provided
+            they have a match in the given dataframe).
+        df: The dataframe to be searched for corresponding columns.
+        allow_missing: Flag controlling if objects are allowed to have no corresponding
+            columns in the dataframe.
+        allow_extra: Flag controlling if the dataframe is allowed to have columns
+            that have corresponding objects.
+
+    Raises:
+        ValueError: If the given objects and dataframe are not compatible
+            under the specified values for the Boolean flags.
+
+    Returns:
+        The (subset of) objects that need to be considered for the transformation.
+    """
+    names = [p.name for p in objects]
+
+    if (not allow_missing) and (missing := set(names) - set(df)):  # type: ignore[arg-type]
+        raise ValueError(
+            f"The object(s) named {missing} cannot be matched against "
+            f"the provided dataframe. If you want to transform a subset of "
+            f"columns, explicitly set `allow_missing=True`."
+        )
+
+    if (not allow_extra) and (extra := set(df) - set(names)):
+        raise ValueError(
+            f"The provided dataframe column(s) {extra} cannot be matched against"
+            f"the given objects. If you want to transform a dataframe "
+            f"with additional columns, explicitly set `allow_extra=True'."
+        )
+
+    return [p for p in objects if p.name in df] if allow_missing else list(objects)
```

Comment:
is this if clause needed?
If `allow_missing` is `False` the check above alerady made sure that all objects are in the df, effetively giving the same result as the list comprehension

Reply:
I guess you are right. Probably an artifact from a previous version (note: the code is not new, just moved it to another location in this PR). Have removed the part at the end